code/game/turfs/simulated/floor/fancy_floor.dm

```diff
@@ -43,7 +43,32 @@
 		new /obj/item/weapon/ore/glass(src)
 		new /obj/item/weapon/ore/glass(src) //Make some sand if you shovel grass
 		to_chat(user, "<span class='notice'>You shovel the grass.</span>")
+		playsound(src, 'sound/effects/shovel_dig.ogg', 50, 1)
 		make_plating()
+		
+// NEEDS TO BE UPDATED
+/turf/simulated/floor/basalt //By your powers combined, I am captain planet
+	name = "volcanic floor"
+	icon = 'icons/turf/floors.dmi'
+	icon_state = "basalt0"
+	oxygen = 14
+	nitrogen = 23
+	temperature = 300
+	
+/turf/simulated/floor/attackby(obj/item/W, mob/user, params)
```

Comment:
Just checked - this takes priority over the normal `floor/attackby`, meaning this would've broken practically all interaction with floors except shoveling basalt out of them. Whoops!